api/src/api/schemas/search_schema.py

```diff
@@ -104,4 +112,59 @@ def with_one_of(
         return self
 
     def build(self) -> Schema:
-        return BaseSearchSchema.from_dict(self.schema_fields, name=self.schema_class_name)  # type: ignore
+        return super().build()
+
+
+class DateSearchSchemaBuilder(BaseSearchSchemaBuilder):
+    """
+    Builder for setting up a filter for a date in the search endpoint schema.
+
+    Example of what this might look like:
+        {
+            "filters": {
+                "post_date": {
+                    "start_date": "YYYY-MM-DD",
+                    "end_date": "YYYY-MM-DD"
+                }
+            }
+        }
+
+    Support for start_date and
+    end_date filters have been partially implemented.
+
+    Usage::
+    # In a search request schema, you would use it like so:
+
+        example_start_date_field = fields.Nested(
+            StrSearchSchemaBuilder("ExampleStartDateFieldSchema")
+                .with_start_date()
+                .build()
+        )
+
+        example_end_date_field = fields.Nested(
+            StrSearchSchemaBuilder("ExampleEndDateFieldSchema")
+                .with_end_date()
+                .build()
+        )
+
+        example_startend_date_field = fields.Nested(
+            StrSearchSchemaBuilder("ExampleStartEndDateFieldSchema")
+                .with_start_date()
+                .with_end_date()
+                .build()
+        )
+    """
+
+    def __init__(self, schema_class_name: str):
+        super().__init__(schema_class_name)
```

Comment:
If you aren't changing anything about the implementation of init and build, you can just exclude them from the subclasses.

Reply:
Cool! I wasn't sure if I could exclude the __init__ method from the subclasses, but this is good to know. Thank you. Done :)